src/ui/window.c

```diff
@@ -1406,9 +1408,7 @@ win_print_incoming(ProfWin* window, const char* const display_name_from, ProfMes
             enc_char = strdup("-");
         }
 
-        if (prefs_get_boolean(PREF_CORRECTION_ALLOW) && message->replace_id) {
-            _win_correct(window, message->plain, message->id, message->replace_id, message->from_jid->barejid);
-        } else {
+        if (!prefs_get_boolean(PREF_CORRECTION_ALLOW) || !message->replace_id || !_win_correct(window, message->plain, message->id, message->replace_id, message->from_jid->barejid)) {
```

Comment:
How about moving those first two checks inside `_win_correct()` and simply always call it?

Reply:
That would make sense.

Reply:
Moved the check, but without prefs_get_boolean, as it would duplicate checking and be inefficient.

Reply:
Same as above, let's continue discussion here https://github.com/profanity-im/profanity/pull/1902#discussion_r1381536586